staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -147,6 +147,28 @@ func Flatten(agg Aggregate) Aggregate {
 	return NewAggregate(result)
 }
 
+// UniqueAggregateWithCount removes duplicates from flattened Aggregate
+// and returns count for each unique error present in the Aggregate
+func UniqueAggregateWithCount(agg Aggregate) Aggregate {
+	if agg == nil || len(agg.Errors()) == 0 {
+		return agg
+	}
+	errs := Flatten(agg).Errors()
```

Comment:
given that you already have the original err[] list from the predicates function you are doing an extra N operations here.  So it's 2N+Y where Y is the reduction.  I'd almost list to change it at the source to bin the errors.  I swear @jayunit100 did just that but it must have been in a different section of code.